erpnext/stock/doctype/pick_list/pick_list.js

```diff
@@ -101,22 +106,34 @@ frappe.ui.form.on('Pick List', {
 		frm.trigger('add_get_items_button');
 	},
 	create_delivery_note: (frm) => {
-		frappe.model.open_mapped_doc({
-			method: 'erpnext.stock.doctype.pick_list.pick_list.create_delivery_note',
-			frm: frm
-		});
+		if (!(frm.doc.locations && frm.doc.locations.length)) {
+			frappe.msgprint(__('Add items in the Item Locations table'));
+		} else {
+			frappe.model.open_mapped_doc({
+				method: 'erpnext.stock.doctype.pick_list.pick_list.create_delivery_note',
+				frm: frm
+			});
+		}
+
 	},
 	create_stock_entry: (frm) => {
-		frappe.xcall('erpnext.stock.doctype.pick_list.pick_list.create_stock_entry', {
-			'pick_list': frm.doc,
-		}).then(stock_entry => {
-			frappe.model.sync(stock_entry);
-			frappe.set_route("Form", 'Stock Entry', stock_entry.name);
-		});
+		if (!(frm.doc.locations && frm.doc.locations.length)) {
+			frappe.msgprint(__('Add items in the Item Locations table'));
+		} else {
```

Comment:
if (!(frm.doc.locations && frm.doc.locations.length)) {
			frappe.msgprint(__('Add items in the Item Locations table'));
} 

Same code used at line no 109, commonified it

Reply:
done. Added `create_delivery_note` and `create_stock_entry` validation on server side itself.